src/components/dialogs/delete-attaching-line-dialog.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import Grid from '@mui/material/Grid';
+import { useParams } from 'react-router-dom';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { useInputForm, useTextValue } from './inputs/input-hooks';
+import {
+    gridItem,
+    GridSection,
+    compareById,
+    sanitizeString,
+} from './dialogUtils';
+import { deleteAttachingLine } from '../../utils/rest-api';
+import PropTypes from 'prop-types';
+import { useAutocompleteField } from './inputs/use-autocomplete-field';
+import ModificationDialog from './modificationDialog';
+
+const getId = (e) => e?.id || (typeof e === 'string' ? e : '');
+
+/**
+ * Dialog to delete attaching line.
+ * @param lineOptionsPromise Promise handling list of network lines
+ * @param currentNodeUuid the currently selected tree node
+ * @param editData record to edit
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+const DeleteAttachingLineDialog = ({
+    lineOptionsPromise,
+    currentNodeUuid,
+    editData,
+    ...dialogProps
+}) => {
+    const studyUuid = decodeURIComponent(useParams().studyUuid);
+
+    const { snackError } = useSnackMessage();
+
+    const inputForm = useInputForm();
+
+    const [formValues, setFormValues] = useState(undefined);
+
+    const [lineOptions, setLineOptions] = useState([]);
+
+    const [loadingLineOptions, setLoadingLineOptions] = useState(true);
+
+    useEffect(() => {
+        if (editData) {
+            setFormValues(editData);
+        }
+    }, [editData]);
+
+    useEffect(() => {
+        if (!lineOptionsPromise) return;
+        lineOptionsPromise.then((values) => {
+            setLineOptions(values);
+            setLoadingLineOptions(false);
+        });
+    }, [lineOptionsPromise]);
+
+    const formValueLineTo1AttachId = useMemo(() => {
+        return formValues?.lineToAttachTo1Id
+            ? { id: formValues?.lineToAttachTo1Id }
+            : { id: '' };
+    }, [formValues]);
+
+    const formValueLineTo2AttachId = useMemo(() => {
+        return formValues?.lineToAttachTo2Id
+            ? { id: formValues?.lineToAttachTo2Id }
+            : { id: '' };
+    }, [formValues]);
+
+    const formValueAttachedLineId = useMemo(() => {
+        return formValues?.attachedLineId
+            ? { id: formValues?.attachedLineId }
+            : { id: '' };
+    }, [formValues]);
+
+    const [lineToAttachTo1, lineToAttachTo1Field] = useAutocompleteField({
+        id: 'lineToAttachTo1',
+        label: 'ID',
+        validation: { isFieldRequired: true },
```

Comment:
the field is marked as mandatory, but it is possible to validate the dialog with an empty value for this field !

Reply:
yes it's mandatory, so the validation gonna fail if the field is empty

Reply:
the validation does not fail for this field.
test case: create a new modification; fill only the optional field : only one field seems to be mandatory.
So we can send an empty value for 'lineToAttachTo1'